### PR TITLE
add user table with migration

### PR DESCRIPTION
**Pull request by msm-faraj**
this pr aims to:

- create user table
- this table has almost all column that any user table capable to have
- we can choose what we need

### CODE REVIEW

File: src/migrations/20240726120857-add-user-table.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      },
+      firstName: {
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      lastName: {
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      profilePictureUrl: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:36]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      phoneNumber: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:40]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      gender: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:44]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      twoFactorSecret: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:48]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      status: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:52]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        allowNull: false,
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.NOW,
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.NOW,
+      },
+      dateOfBirth: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:74]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+      },
+      lastLogin: {
+        type: Sequelize.DATE,
+        allowNull: true,
+      },
+      deletedAt: {
+        type: Sequelize.DATE,
+        allowNull: true,
+      },
+      address: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:86]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        type: Sequelize.JSON,
+        allowNull: true,
+      },
+      preferences: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:90]: Not needed

File: src/migrations/20240726120857-add-user-table.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+        type: Sequelize.JSON,
+        allowNull: true,
+      },
+      bio: {
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:94]: Not needed

Comment (msm-faraj) [reply src/migrations/20240726120857-add-user-table.js]: done

**Comment by msm-faraj** [comment]
All of the not-needed columns have been deleted

File: src/migrations/20240726120857-add-user-table.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        allowNull: true,
+      },
+    });
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropTable("Users");
+    await queryInterface.sequelize.query(
+      'DROP TYPE IF EXISTS "enum_Users_status";'
+    );
```
Comment (pouyapourbaba) [inline src/migrations/20240726120857-add-user-table.js:67]:
```suggestion
```

Not required anymore

Comment (msm-faraj) [reply src/migrations/20240726120857-add-user-table.js]: done

**Review by pouyapourbaba — APPROVED** [review]

**Pull request merged**